### PR TITLE
add Fortran API for functionspace::CellColumns

### PR DESCRIPTION
**Pull request by sbrdar**

### CODE REVIEW

File: src/atlas/functionspace/detail/CellColumnsInterface.cc [hunk 1]
```diff
@@ -0,0 +1,1134 @@
...
+    ATLAS_ASSERT(This != nullptr, "Cannot access uninitialised atlas_functionspace_CellColumns");
+    ATLAS_ASSERT(field != nullptr, "Cannot access uninitialised atlas_Field");
+    std::string checksum_str(This->checksum(field));
+    size      = static_cast<int>(checksum_str.size());
+    checksum  = new char[size + 1];
+    allocated = true;
+    std::strncpy(checksum, checksum_str.c_str(), size + 1);
+}
+
+/*
```
Comment (wdeconinck) [inline src/atlas/functionspace/detail/CellColumnsInterface.cc:151]: Please delete commented code.

File: src/atlas_f/functionspace/atlas_functionspace_CellColumns_module.fypp [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+! This software is licensed under the terms of the Apache Licence Version 2.0
+! which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+! In applying this licence, ECMWF does not waive the privileges and immunities
+! granted to it by virtue of its status as an intergovernmental organisation nor
+! does it submit to any jurisdiction.
+
+#include "atlas/atlas_f.h"
+#:include "internals/atlas_generics.fypp"
+
+#:set ranks = [0,1]
```
Comment (wdeconinck) [inline src/atlas_f/functionspace/atlas_functionspace_CellColumns_module.fypp:12]: Please remove, and also rename file with ".F90" extension. No fypp is used here.

**Review by wdeconinck — COMMENTED** [review]
Thanks @sbrdar this was missing indeed!
A few cosmetic remarks to address.

File: src/atlas/functionspace/detail/CellColumnsInterface.cc (hunk 1 above)
Comment (sbrdar) [reply src/atlas/functionspace/detail/CellColumnsInterface.cc]: Thanks! Removed.

File: src/atlas_f/functionspace/atlas_functionspace_CellColumns_module.fypp (hunk 2 above)
Comment (sbrdar) [reply src/atlas_f/functionspace/atlas_functionspace_CellColumns_module.fypp]: Thanks! Done.

**Pull request merged**